### PR TITLE
Imgui bundle fixes

### PR DESCRIPTION
**Pull request by pthom**
Hi,

Congrats for this nice library!

While integrating your library in a library on which I am currently working, I encountered two small issues, for which I did a fix. 

PS: I was very impressed by your  AeroToy demo on youtube! Nice.

I'll contact you in PM in a few weeks about the library I'm working on, in order to keep you informed.

Quick explanations about the fixes:

* 54387426a136849389757606a493e12ffd95233c: texture2d was deprecated and replaced by texture
* 3e3f2b9eec848ca85ad21e523e6ccd249e0d2a58: ImTextureID is by default typefed to void*, but it is a user customization point

### CODE REVIEW

**Comment by andyborrell** [comment]
Hey thanks for this.  Will try to take a look as soon as I can.  Unfortunately I'm busy with other life stuff at the moment, but will hopefully be able to take a look soon.

**Comment by dcnieho** [comment]
I can confirm that this fixes an error i encountered on Windows 10:

```
ERROR: ImGui_ImplOpenGL3_CreateDeviceObjects: failed to compile fragment shader!
ERROR: 1:23: 'function' : is removed in Forward Compatible context texture2D
ERROR: 1:23: 'texture2D' : no matching overloaded function found (using implicit conversion)
ERROR: 1:23: 'texture2D' : function is not known
ERROR: 1:23: '+' : wrong operand types no operation '+' exists that takes a left-hand operand of type '4X4 matrix of highp float' and a right operand of type 'uniform 4-component vector of highp float' (or there is no acceptable conversion)
ERROR: 1:23: '=' :  cannot convert from '4X4 matrix of highp float' to '4-component vector of highp float'


ERROR: ImGui_ImplOpenGL3_CreateDeviceObjects: failed to link shader program! (with GLSL '#version 130
')
Attached fragment shader is not compiled.
```

File: backends/tex_inspect_opengl.cpp [hunk 1]
```diff
@@ -387,7 +387,7 @@ const GLchar *fragment_shader_glsl_120 = "#ifdef GL_ES\n"
                                          "        uv = Frag_UV;\n"
                                          "    vec2 texelEdge = step(mod(texel,vec2(1.0,1.0)),GridWidth);\n"
                                          "    float isGrid = max(texelEdge.x, texelEdge.y);\n"
-                                         "    vec4 ct = ColorTransform * texture2D(Texture, uv) + ColorOffset;\n"
+                                         "    vec4 ct = ColorTransform * texture(Texture, uv) + ColorOffset;\n"
```
Comment (andyborrell) [inline backends/tex_inspect_opengl.cpp]: There is no texture function in GLSL 1.20, we need to stick with texture2D here.

**Comment by andyborrell** [comment]
This looks good.  I don't think we need to change the GLSL 1.20 fragment shader code though.  I checked imgui's own backend code and they do use `texture` for everything except 1.20.

**Comment by pthom** [comment]
> This looks good. I don't think we need to change the GLSL 1.20 fragment shader code though. I checked imgui's own backend code and they do use `texture` for everything except 1.20.

You are right, I fixed this

**Pull request merged**